components/channel/cmd/main.go

```diff
@@ -40,42 +40,44 @@ func main() {
 	// Log Environment Variables
 	logger.Info("Environment Variables",
 		zap.String("METRICS_PORT", metricsPort),
-		zap.String("HTTP_PORT", httpPort),
 		zap.Any("KAFKA_BROKERS", kafkaBrokers),
 		zap.String("KAFKA_TOPIC", kafkaTopic),
 		zap.String("CLIENT_ID", clientId),
 		zap.String("KAFKA_USERNAME", kafkaUsername),
 		zap.String("KAFKA_PASSWORD", kafkaPassLog))
 
 	// Validate Required Environment Variables
-	if len(httpPort) == 0 ||
-		len(metricsPort) == 0 ||
+	if len(metricsPort) == 0 ||
 		len(kafkaBrokers) == 0 ||
 		len(kafkaTopic) == 0 ||
 		len(clientId) == 0 {
 		logger.Fatal("Invalid / Missing Environment Variables - Terminating")
 	}
 
 	ctx := context.Background()
-	port, err := strconv.Atoi(httpPort)
-	if err != nil {
-		logger.Error("Invalid HTTP port specified, could not be converted to int", zap.Error(err))
-		return
-	}
 
-	transport, err := cloudevents.NewHTTPTransport(
-		cloudevents.WithPort(port),
-		cloudevents.WithPath("/"),
-		cloudevents.WithEncoding(cloudevents.HTTPBinaryV03),
-	)
+	// The EventReceiver is responsible for processing the context
+	// (headers and binary/json content) of each request and then
+	// passing the context, channel details, and the constructed
+	// CloudEvent event to our handleEvent function.
+	eventReceiver, err := eventingChannel.NewEventReceiver(handleEvent, logger)
 	if err != nil {
-		logger.Error("failed to create transport", zap.Error(err))
+		logger.Error("failed to create the event_receiver", zap.Error(err))
 		return
 	}
 
-	cloudEventClient, err := cloudevents.NewClient(transport)
+	// The Knative CloudEvent Client handles the mux http server
+	// setup (middlewares and transport options) and invokes
+	// the eventReceiver. Althought the NewEventReceiver method
+	// above will also invoke kncloudevents.NewDefaultClient
+	// internally, that client goes unused when using the ServeHTTP
+	// on the eventReceiver.
+	// IMPORTANT: Because the kncloudevents package does not allow
+	// injecting modified configuration, we can't override the
+	// default port being used, 8080.
+	knCloudEventClient, err := kncloudevents.NewDefaultClient()
```

Comment:
Assuming there are changes later on changing the channel service port configuration in the controller?